app/controllers/concerns/account_recoverable.rb

```diff
@@ -1,5 +1,5 @@
 module AccountRecoverable
   def piv_cac_enabled_but_not_phone_enabled?
-    current_user.piv_cac_enabled? && !current_user.phone_enabled?
+    current_user.piv_cac_enabled? && !current_user.phone_configuration&.mfa_enabled?
```

Comment:
Given that we know this logic will change for sure when we support multiple phone configurations, what do you think about encapsulating this logic inside the `SmsLoginOptionPolicy` class, or some other Policy class? That way, when the logic inevitably changes, we will only need to make a change in one place.

Reply:
Same comment about other places we call something like `user.phone_configuration.[some_other_method]`. Perhaps a `UserWithPhonePolicy` that can define the various methods needed, like `mfa_phone`, `delivery_preference`, `mfa_enabled?`

Reply:
Longer term, some of this makes sense. This PR is focused on shifting the reading from the `users` table to the `phone_configurations` table. I'd rather not muddy it by shifting a lot of code to different places.

Once we shift to multiple phones, it won't be as simple as `UserWithPhonePolicy` since we'll be asking things about either a particular phone (e.g., is _this_ phone enabled for mfa?) or about the collection of phones (e.g., is _any_ phone enabled for mfa?). I'm not sure we buy a lot by wrapping an object in another object just to ask the same questions and have them all passed through to the wrapped object.

Reply:
Let me try another way of illustrating the issue. In multiple places, we have replaced `current_user.phone_enabled?` with `current_user.phone_configuration&.mfa_enabled?`, and at the same time, we have updated the definition of `phone_enabled?` to be `phone_configuration&.mfa_enabled?`. Why not just make that one change and keep all the other calls to `current_user.phone_enabled?`

Same with calls to `current_user.phone` and `current_user.otp_delivery_preference`. Why not define those in a single place to keep things DRY?

Reply:
The reason for redefining the methods in `user` was to catch any cases where I might have missed something. Even if I get an error later when I remove those methods, it'll still be getting the data from the new table. Once we move to multiple phones, those methods will have to go away anyway.

The  `otp_delivery_preference` isn't redefined in the `user` model because we have places where that is set without the `phone` being set. So there are cases where the OTP delivery preference isn't tied to a phone number.

Reply:
That's why I was suggesting a Policy class, so that you don't need to override User methods. When we write `current_user.phone_configuration&.mfa_enabled?`, what are we asking? We are asking, "does the current user have at least one phone that is configured to receive an OTP via SMS or Voice?"

Once we support multiple phone numbers, that question will remain the same. However, the way we answer it will be different. When that time comes, would you rather have only one place where that new query needs to be updated, or would you prefer to have to update it in 10 different places? If you were reviewing such a code change, which change would be easier to review?

Reply:
Definitely. I just don't want to expand the scope of this PR. This PR is only about reading from the phone configurations table rather than the users table for phone information.

Reply:
Ah, I see. We agree, but not on whether this goes in this PR. If it were me, I would include it in this PR because it's a related change and only adds one file. Alternatively, the change could be in a follow-up PR shortly after this one is merged, but I leave it up to you.

Reply:
Yup. I plan to do some of that as part of the PRs that add multi-phone capabilities and the ability to turn off phone use for MFA starting next week.